01-data-structures/06-trees/min_binary_heap/min_binary_heap.rb

```diff
@@ -0,0 +1,127 @@
+require_relative "node"
+
+class MinBinaryHeap
+  attr_reader :root
+  attr_reader :tree
+
+  def initialize(root)
+    @root = root
+    @tree = [nil, @root]
+  end
+
+  def insert(root, data)
+    @tree.push(data)
+    index = @tree.length - 1
+    parent = index / 2
+
+    is_data_a_left_child = @tree[parent] && index % 2 == 0
+
+    if is_data_a_left_child
+      @tree[parent].left = @tree[index]
+    else
+      @tree[parent].right = @tree[index]
+    end
+
+    while is_parent_greater_than_child(index, parent)
+      temp_title = @tree[parent].title
+      temp_rating = @tree[parent].rating
+      @tree[parent].title = @tree[index].title
+      @tree[parent].rating = @tree[index].rating
+      @tree[index].title = temp_title
+      @tree[index].rating = temp_rating
+      index = parent
+      parent = index / 2
+    end
+  end
+
+  def find(root, data)
+    (1...@tree.length).each do |location|
+      if data.title == @tree[location].title
+        return @tree[location]
+      end    
+    end
+    nil
+  end
+
+  def  is_parent_greater_than_child(index, parent)
+    @tree[parent] && @tree[parent].rating > @tree[index].rating
```

Comment:
shouldn't we be checking for @tree[parent] before this? otherwise we don't even need to callt his